xtask/src/ci.rs

```diff
@@ -95,4 +102,77 @@ impl CiTask {
 
         fmt_res.and(check_full_res).and(clippy_res).and(sort_res)
     }
+
+    fn check_spec_links(&self) -> Result<()> {
+        let mut start_path = self.project_root.clone();
+        start_path.push("crates");
+        walk_dirs(start_path)
+    }
+}
+
+fn walk_dirs<P: AsRef<Path>>(path: P) -> Result<()> {
```

Comment:
Can you move this to a separate child module? It accounts for ~~almost half~~ more than half (after other commits) the lines of code despite being only a very small part of the checks.